webapi/src/main/java/org/sakaiproject/webapi/controllers/CalendarController.java

```diff
@@ -97,40 +100,73 @@
                 }
             }
         } catch (Exception e) {
-            log.warn("Failed to create bean for calendar event");
+            log.warn("Failed to create bean for calendar event: {}", e.toString());
         }
         return bean;
     };
 
-	@GetMapping(value = "/users/{userId}/calendar", produces = MediaType.APPLICATION_JSON_VALUE)
-    public List<CalendarEventRestBean> getUserCalendar(@PathVariable String userId) throws UserNotDefinedException {
+    @GetMapping(value = "/users/current/calendar", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, Object> getCurrentUserCalendar() throws UserNotDefinedException {
 
-		Session session = checkSakaiSession();
+        checkSakaiSession();
 
-        return siteService.getUserSites().stream().map(s -> {
+        Instant projectCutoff = Instant.now().minus(serverConfigurationService.getInt("webapi.calendar.project_site_age_in_days", 730), ChronoUnit.DAYS);
 
-            try {
-                return ((List<CalendarEvent>) calendarService.getCalendar(calendarService.calendarReference(s.getId(), "main"))
-                    .getEvents(null, null))
```

Comment:
It would be a good idea to move the filtering into the service method call so that it can be used more broadly.